plugin/trino-starrocks/src/main/java/io/trino/plugin/starrocks/StarRocksClient.java

```diff
@@ -0,0 +1,676 @@
+    }
+
+    @Override
+    public Optional<String> getTableComment(ResultSet resultSet)
+    {
+        // Don't return a comment until the connector supports creating tables with comment
+        return Optional.empty();
+    }
+
+    @Override
+    protected String getTableSchemaName(ResultSet resultSet)
+            throws SQLException
+    {
+        return resultSet.getString("TABLE_CAT");
+    }
+
+    @Override
+    protected List<String> createTableSqls(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    {
+        if (tableMetadata.getComment().isPresent()) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables with table comment");
+        }
+
+        Map<String, Object> tableProperties = tableMetadata.getProperties();
+        StarRocksEngineType engine = StarRocksTableProperties.getEngine(tableProperties);
+        List<String> distributionDesc = StarRocksTableProperties.getDistributionDesc(tableProperties);
+        // OLAP engine type table requires distribution keys
+        if (engine == StarRocksEngineType.OLAP && distributionDesc.isEmpty()) {
+            List<ColumnMetadata> columnsMetadata = tableMetadata.getColumns();
+            checkArgument(!columnsMetadata.isEmpty(), "Can not get column metadata");
+            distributionDesc = ImmutableList.of(columnsMetadata.iterator().next().getName());
+        }
+
+        String createTableSql = "CREATE TABLE %s (%s) \n ENGINE = %s".formatted(
+                quoted(remoteTableName),
+                join(", ", columns),
+                engine.getEngineType());
+
+        if (!distributionDesc.isEmpty()) {
+            verifyDistributionDescExistsInTable(tableMetadata, distributionDesc);
+            createTableSql += distributionDesc.stream()
+                    .map(this::quoted)
+                    .collect(joining(", ", "\n DISTRIBUTED BY HASH (", ")"));
+        }
+
+        createTableSql += buildCreateTableProperties(engine, tableProperties);
+
+        return ImmutableList.of(createTableSql);
+    }
+
+    private String buildCreateTableProperties(StarRocksEngineType engineType, Map<String, Object> tableProperties)
+    {
+        String tablePropertiesFormat = "\n PROPERTIES (%s)";
+        ImmutableList.Builder<String> tablePropertiesBuilder = ImmutableList.builder();
+        if (engineType == StarRocksEngineType.OLAP) {
+            int replicationNum = StarRocksTableProperties.getReplicationNum(tableProperties);
+            checkArgument(replicationNum > 0, "Replication number must greater than 0");
+            tablePropertiesBuilder.add("\"%s\" = \"%d\"".formatted(REPLICATION_NUM, replicationNum));
+
+            return tablePropertiesFormat.formatted(join(", ", tablePropertiesBuilder.build()));
+        }
+
+        return "";
+    }
+
+    private void verifyDistributionDescExistsInTable(ConnectorTableMetadata tableMetadata, List<String> distributionDesc)
```

Comment:
Move this above `buildCreateTableProperties` (the order in which they are invoked) - easier to read/jump in code.